libfwupdplugin/fu-device.h

```diff
@@ -493,6 +493,20 @@ typedef guint64 FuDeviceInternalFlags;
  */
 #define FU_DEVICE_INTERNAL_FLAG_NO_PROBE_COMPLETE (1ull << 27)
 
+/**
+ * FU_DEVICE_INTERNAL_FLAG_RETAIN_FIRMWARE_ARCHIVE:
+ *
+ * Save the cabinet archive to persistent storage before starting the update process.
+ *
+ * This is super useful when the network device is being updated, and different blobs inside the
+ * archive could be required in different scenarios. For instance, if the user installs a firmware
+ * update for a specific network device and then changes the SIM -- it might be we need the archive
+ * again and have no internet access.
+ *
+ * Since: 1.8.13
+ */
+#define FU_DEVICE_INTERNAL_FLAG_RETAIN_FIRMWARE_ARCHIVE (1ull << 28)
```

Comment:
How about `CACHE_FIRMWARE_ARCHIVE`

Reply:
It's not a cache -- which implies you can just nuke it without consequence. Maybe `BACKUP_FIRMWARE_ARCHIVE`?

Reply:
Maybe _BACKUP_ARCHIVE_REMOTE or _ARCHIVE_BACKUP_REMOTE or _USE_BACKUP_REMOTE or _SAVE_BACKUP_REMOTE perhaps?

Reply:
I think _SAVE_BACKUP_REMOTE is my favorite

Reply:
_SAVE_INTO_BACKUP_REMOTE is even better, thanks. Fixed.